jobs/integration/utils.py

```diff
@@ -343,41 +343,38 @@ async def log_snap_versions(model, prefix="before"):
 async def validate_storage_class(model, sc_name, test_name):
     control_plane = model.applications["kubernetes-control-plane"].units[0]
     # write a string to a file on the pvc
-    pod_definition = """
+    pod_definition = f"""
 apiVersion: v1
 kind: PersistentVolumeClaim
 metadata:
-  name: {0}-pvc
-  annotations:
-   volume.beta.kubernetes.io/storage-class: {0}
+  name: {sc_name}-pvc
 spec:
   accessModes:
   - ReadWriteOnce
+  storageClassName: {sc_name}
```

Comment:
quoting https://kubernetes.io/docs/concepts/storage/dynamic-provisioning/#using-dynamic-provisioning

> Users request dynamically provisioned storage by including a storage class in their PersistentVolumeClaim. Before Kubernetes v1.6, this was done via the volume.beta.kubernetes.io/storage-class annotation. However, this annotation is deprecated since v1.9. Users now can and should instead use the storageClassName field of the PersistentVolumeClaim object. The value of this field must match the name of a StorageClass configured by the administrator